server_fn_macro/src/lib.rs

```diff
@@ -148,36 +173,47 @@ pub fn server_macro_impl(
         );
     };
 
+    let server_ctx_path = if let Some(ctx) = &server_context {
+        let path = &ctx.path;
+        quote!(#path)
+    } else {
+        quote!(())
+    };
+
+    let server_fn_path = server_fn_path
+        .map(|path| quote!(#path))
+        .unwrap_or_else(|| quote! { server_fn });
+
     Ok(quote::quote! {
         #[derive(Clone, Debug, ::serde::Serialize, ::serde::Deserialize)]
         pub struct #struct_name {
             #(#fields),*
         }
 
-        impl leptos::ServerFn for #struct_name {
+        impl #server_fn_path::ServerFn<#server_ctx_path> for #struct_name {
             type Output = #output_ty;
 
             fn prefix() -> &'static str {
                 #prefix
             }
 
             fn url() -> &'static str {
-                #url
+                #server_fn_path::const_format::concatcp!(#fn_name_as_str, #server_fn_path::xxhash_rust::const_xxh64::xxh64(concat!(env!("CARGO_MANIFEST_DIR"), ":", file!(), ":", line!(), ":", column!()).as_bytes(), 0))
```

Comment:
We had some concerns from people about leaking the structure of their app in server function names, but it looks like this is a good way of obscuring that while maintaining unique IDs per source file and avoiding collisions — Thanks! I'm assuming this is not (easily?) reversible.

Reply:
It is not a cryptographic hash, but it should be lossy so you cannot recover the path from the hash. My main reason for doing this approach was it works on stable, but not exposing the file path is a nice bonus.